server/src/test/java/io/crate/planner/optimizer/symbol/OptimizerTest.java

```diff
@@ -43,4 +45,24 @@ public void test_like_on_numeric_columns_keeps_cast_around_reference() throws Ex
             SymbolMatchers.isLiteral("10")
         ));
     }
+
+    @Test
+    public void test_comparison_of_string_and_numeric_doesnt_optimize_away_cast() throws Exception {
+        SQLExecutor e = SQLExecutor.builder(clusterService)
+            .addTable("create table tbl (strCol string, intCol int)")
+            .build();
+
+        //checking either sides of num <-> str comparison, reference casting remains
+        Symbol symbol1 = Optimizer.optimizeCasts(e.asSymbol("strCol::bigint > 3"), e.getPlannerContext(clusterService.state()));
+        Symbol symbol2 = Optimizer.optimizeCasts(e.asSymbol("intCol::string > '3'"), e.getPlannerContext(clusterService.state()));
+
+        assertThat(symbol1, SymbolMatchers.isFunction(GtOperator.NAME));
+        assertThat(symbol2, SymbolMatchers.isFunction(GtOperator.NAME));
+
+        Function func1 = (Function) symbol1;
+        Function func2 = (Function) symbol1;
+        // In all cases below ref cast used to be swapped with parameter.
+        assertThat(func1.arguments().get(0), SymbolMatchers.isFunction("cast"));
+        assertThat(func2.arguments().get(0), SymbolMatchers.isFunction("cast"));
```

Comment:
minor: wouldn't something like that be a bit simpler?
```
assertThat(symbol2, SymbolMatchers.isFunction(
            GtOperator.NAME,
            SymbolMatchers.isFunction("cast"),
            SymbolMatchers.isLiteral("3")
        ));
```

Reply:
yep, changed that, thanks. 

I wanted to check that precisely `arg(0)/ ref` is being casted, had an impression, that `isFunction` just calls contains which could lead to false test success in `strCol::bigint > 3::bigint ` even with old code without fix. I reverted fix locally and test this assumption, I was wrong, so committing simpler assertion